### DISCUSSION TITLE
How to remove Windows GUI Commit ?

### DISCUSSION THREAD

**Discussion by gz1968** [community/community#24895] [Apps, API and Webhooks]
<p>Hello,</p>
<p>I use Windows 10 and the latest GitHub Desktop for Windows.</p>
<p>I tried committing a file over 100 MB.  It was a video, that I had forgotten to shrink.</p>
<p>Now I’m stuck. </p>
<p>I tried reversing commits in Windows GUI.</p>
<p>I’ve uninstalled and reinstalled GitHub desktop.  The commit is still there and stuck.</p>
<p>I’ve installed GIT for the command line.</p>
<p>I try command git log in bash.</p>
<p>$ git log<br>
fatal: not a git repository (or any of the parent directories): .git</p>
<p>I have not idea where this git log is.  I’m a complete Newby and had zero desire to delve into this literature.  But not I have to.</p>
<p>How do I clear out this Commit?</p>

**Comment by Wabri** [comment]
<p>It’s still there because there is the .git folder inside the project. To delete the commit you have to the previous commit. Let’s make this in order…</p>
<p><strong>Command line method</strong></p>
<p>First of all you have to move in the project folder, than open here the git bash command line and type:</p>
<pre><code>git log
</code></pre>
<p>If you are in the right folder you see something like this:</p>
<pre><code>commit 25d34209da48d9a5fd0b2540c7f23b25adc402a3
Author: Wabri &lt;gabriele.puliti@gmail.com&gt;
Date: Thu Oct 25 15:40:11 2018 +0200

    update README

commit 116c0167a589021a12b81a3502b58fe99bbab87f
Author: Wabri &lt;gabriele.puliti@gmail.com&gt;
Date: Wed Oct 24 17:15:11 2018 +0200

    update readme and debinstall

commit eb43dd801ed3d58a776c3cbe8eb6c64d5ce92e35
Author: Wabri &lt;gabriele.puliti@gmail.com&gt;
Date: Sun Oct 21 23:30:43 2018 +0200

    Some update

commit dd197684199a69356ff1efdcebffd378291db7ab
Author: Wabri &lt;gabriele.puliti@gmail.com&gt;
Date: Sun Oct 21 23:26:04 2018 +0200
</code></pre>
<p>In my case if I want to delete the last commit (if and only if I have not pushed yet), i have to take the token of the previous commit:</p>
<pre><code>commit&lt;u&gt;&lt;strong&gt;116c0167a589021a12b81a3502b58fe99bbab87f&lt;/strong&gt;&lt;/u&gt;&lt;----------- this one
Author: Wabri &lt;gabriele.puliti@gmail.com&gt;
Date: Wed Oct 24 17:15:11 2018 +0200

    update readme and debinstall
</code></pre>
<p>Then move back in time with reset, like this:</p>
<pre><code>git reset --hard 116c0167a589021a12b81a3502b58fe99bbab87f
</code></pre>
<p><strong> </strong></p>
<p>From here you can create a branch or create a new commit to push (if and only if you have not pushed yet).</p>
<p><strong>Github GUI method</strong></p>
<p>Dont really know, i dont use that, sorry</p>
<p>I hope I have solved your problem</p>
<p>-Gabriele-</p>

**Comment by steveward** [comment, marked as answer]
<p><a class="mention" href="/gz1968">@gz1968</a> we are in the process of <a href="https://github.com/desktop/desktop/pull/6013" rel="nofollow noopener">shipping a feature to warn about committing files over 100MB</a> in GitHub Desktop, which should hopefully prevent these types of issues from occurring in the future. Currently it isn’t possible to remove a large file in GitHub Desktop, but we do have a <a href="https://help.github.com/articles/removing-files-from-a-repository-s-history/" rel="nofollow noopener">help article</a> that walks through some of the available options for removing files from your repository’s history. You will need to use the command line for these steps.</p>
<p>If you follow the steps that <a class="mention" href="/wabri">@wabri</a> shared please note that the <code>--hard</code> flag is destructive in nature – it will delete all of the files in the commit where you committed the large file. I would recommend using the <code>--mixed</code> flag since that will leave your other files untouched in your working directory. You can then right-click on the file in GitHub Desktop and select <code>Discard changes...</code> to delete the specific large file that you created. I hope that is helpful!</p>
